general/forall.hpp

```diff
@@ -35,10 +35,15 @@ namespace mfem
 
 // The MFEM_FORALL wrapper
 #define MFEM_FORALL(i,N,...)                                     \
-   ForallWrap(N,                                                 \
+   ForallWrap(false,N,                                           \
               [=] MFEM_ATTR_DEVICE (int i) {__VA_ARGS__},        \
               [&]                  (int i) {__VA_ARGS__})
 
+// The MFEM_FORALL_IF wrapper
+#define MFEM_FORALL_IF(locked,i,N,...)                      \
```

Comment:
```suggestion
#define MFEM_FORALL_IF(use_cpu,i,N,...)                      \
```

Reply:
I would even change `use_cpu` to `use_dev` since it's always use like `MFEM_FORALL_IF(!use_dev,...)`, the `!` everywhere is strange.

Reply:
Good point. I'll invert the logic and rename the parameter to `use_dev`.